File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num: nums) {
            set.add(num);
        }

        int lcs = 0;
        int n = nums.length;
        for(int i=0; i<n; i++) {
            if(set.contains(nums[i] + 1)) {
                continue;
            }   

            int count = 1;
            while(set.contains(nums[i] - count)) {
                count++;
            }

            lcs = Math.max(lcs, count);
        }

        return lcs;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        lcs = 0
        nums = set(nums)
        for num in nums:
            if num - 1 in nums:
                continue
            else:
                length = 1
                while num + length in nums:
                    length += 1
                
                lcs = max(lcs, length)
        
        return lcs
